### PR TITLE
csi: add csi-addon sidecar to the rbd provisioner deployment

### PR DESCRIPTION
**Pull request by rewantsoni**
RBD Provisioner was missing csi-addons sidecar, adding it in this PR

Ref for CSI-Addons Sidecar: [container config](https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/template/rbd/csi-rbdplugin-provisioner-dep.yaml#L140-L180)

### CODE REVIEW

**Comment by rewantsoni** [comment]
/hold
testing is in progress

File: pkg/templates/csisidecars.go [hunk 1]
```diff
@@ -143,6 +143,67 @@ var SnapshotterContainer = &corev1.Container{
...
+	Name: "csi-addons",
+	Args: []string{
+		"--node-id=$(NODE_ID)",
+		"--v=5",
+		fmt.Sprintf("--csi-addons-address=%s", DefaultCSIAddonsSocketPath),
+		fmt.Sprintf("--controller-port=%v", DefaultCSIAddonsContainerPort),
+		"--pod=$(POD_NAME)",
+		"--namespace=$(POD_NAMESPACE)",
+		"--pod-uid=$(POD_UID)",
+		fmt.Sprintf("--stagingpath=%s", DefaultStagingPath),
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply pkg/templates/csisidecars.go:156]: the default value exists in other sidecar containers use that one and yes we need to have leader-election when running with provisioner pod

[the remark this answers is not in the document]
Comment (rewantsoni) [reply pkg/templates/csisidecars.go:156]: Discussed Offline and updated the leader-election-namespace for the csi-addons sidecar and opened a [issue](https://github.com/red-hat-storage/ocs-client-operator/issues/86) to set the values for the other flags in all the sidecars using this flag

**Comment by leelavg** [comment]
@rewantsoni just curious, how did this issue surfaced?

**Comment by rewantsoni** [comment]
> @rewantsoni just curious, how did this issue surfaced?

For RDR, two sidecars were missing in the provisioner csi-addons and csi-omap-generator, after a discussion with @Madhu-1  he mentioned that csi-addons should be deployed by default and omap should be deployed when we are enabling DR

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**